src/app/crypto/services/pgp.service.ts

```diff
@@ -15,53 +19,95 @@ export async function generateNewKeys(): Promise<{
     curve: 'ed25519',
   });
 
+  const kem = await kemBuilder();
+  const { publicKey: publicKyberKey, privateKey: privateKyberKey } = await kem.keypair();
+
   return {
     privateKeyArmored: privateKey,
     publicKeyArmored: Buffer.from(publicKey).toString('base64'),
     revocationCertificate: Buffer.from(revocationCertificate).toString('base64'),
+    publicKyberKeyBase64: Buffer.from(publicKyberKey).toString('base64'),
+    privateKyberKeyBase64: Buffer.from(privateKyberKey).toString('base64'),
   };
 }
+export function XORhex(a: string, b: string): string {
+  let res = '',
+    i = a.length,
+    j = b.length;
+  while (i-- > 0 && j-- > 0) res = (parseInt(a.charAt(i), 16) ^ parseInt(b.charAt(j), 16)).toString(16) + res;
```

Comment:
if the lengths of the chains are different, this will end before they have travelled the longest chain. Is this the desired behaviour?

Reply:
Inputs should always be of identical length

Reply:
Better to handle the possible undesired case where the length of the strings is not identical in order to save possible problems. 
If you control this by throwing an error as discussed above, this problem would be solved

Reply:
@CandelR I've applied your suggestions